_pytest/assertion/rewrite.py

```diff
@@ -179,11 +179,12 @@ def mark_rewrite(self, *names):
         The named module or package as well as any nested modules will
         be rewritten on import.
         """
-        already_imported = set(names).intersection(set(sys.modules))
-        if already_imported:
-            for name in already_imported:
-                if name not in self._rewritten_names:
-                    self._warn_already_imported(name)
+        already_imported = (
+            (set(names) & set(sys.modules)) - set(self._rewritten_names))
```

Comment:
the python guide on sets proposes to write this as something more like

`set(names).intersection(sys.modules).difference(self._rewritten_names)` it would be nice to use that for the cleanup